### ISSUE TITLE
Issue in rendering tracks while using http url

### ISSUE THREAD

**Issue by namratasp20** [igvteam/igv.js#1348]
Currently I am providing http links for fastaURL, indexURL, track urls.
However I receive the following message and tracks are not displayed - 
**Error accessing resource: http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.bed Status: 0: http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.bed**

I am using following http link to for providing urls ->  **http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/**
eg.
fastaURL : http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.fasta
indexURL : http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.fasta.fai
track url : http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.gff ; http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.bed

Could you help me understand why http link is having error in accessing resource , despite igv.js having support for http and https protocol

**Comment by jrobinso** [comment]
igv.js does not have anything to do with http or https protocol, that is enforced at the level of the web browser itself.   If your page is loaded over https:// you cannot load any data or links over http://, this is forbidden.   Do a web search, this is basic web protocol rules it has nothing to do with igv.js.

**Comment by namratasp20** [comment]
My webpage is loaded over http. The links I am using to load data are also over http.

**Comment by jrobinso** [comment]
You mentioned elsewhere "deployed via https", or something like that, so I guessed that was the problem.   As it is I don't know, if you think its an igv.js problem specifically I will probably need a reproducible test case I can run.

**Comment by jrobinso** [comment]
This looks like a duplicate of #1347, is that correct.   As I suggested there, it might be helpful for you to examine the network requests and responses in a web browser development tool, such as the ones built into Chrome and Safari,  as well as look for any error messages in the console.    Also, is CORS support enabled on the servers you are trying to access?

**Comment by namratasp20** [comment]
Thanks for the response

I checked for browser console message - **Access to XMLHttpRequest at 'http://pallab.cds.iisc.ac.in/INTERPIN/Ecoli/ex_hp.bed' from origin 'http://localhost:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.**

However I dont get above error if I am trying to access .fasta files placed in above url. This issue is arising only for .gff and .bed files.

**Comment by namratasp20** [comment]
**Below is the script I am using in my html code for igv **
<script type="text/javascript">

    $(document).ready(function () {

       var options =
        {
        reference:{
                "id": "Bordetella pertussis",
                "name": "Bordetella pertussis",
                "fastaURL": "http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.fasta",
                "indexURL": "http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.fasta.fai",
                "tracks": [
                   {
                    "name": "all hp",
                    "url": "http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.bed",
                    gffTags: "on",
                    color: "green",
                    altColor: "brown",
                   },
                   {
                    "name": "operon",
                    "url": "http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/operon.bed",
                    gffTags: "on",
                    color: "orange",
                    altColor: "rgb(200, 0, 100)",
                   },
                   {
                    "name": "Bordetella pertussis",                   
                    "url": "http://pallab.cds.iisc.ac.in/INTERPIN/Bordtella/NZ_CP010265.gff",
                    "height":300,
                    altColor: "red",
                    "searchable": "True",
                   }
               ]
            }

        };

        var igv_custom_track_click = $('#igv-custom-track-click');

        igv.createBrowser(igv_custom_track_click.get(0), options)

                .then(function (browser) {

                    var genesInList = {};

                    browser.on('trackclick', function (track, popoverData) {
                        var symbol = null;
                        popoverData.forEach(function (nameValue) {
                            if (nameValue.name && nameValue.name.toLowerCase() === 'name') {
                                symbol = nameValue.value;
                            }
                        });

                        if (symbol && !genesInList[symbol]) {
                            if (symbol.indexOf('fc_')!==-1 || symbol.indexOf('fs_')!==-1 || symbol.indexOf('rc_')!==-1 || symbol.indexOf('rs_')!==-1) {
                            genesInList[symbol] = true;
                            $("#geneList").append('<li><a href="/result/hp/Bordetella pertussis/' + symbol +'">' + symbol + '</a></li>');
                           }
                        }

                        // Prevent default pop-over behavior
                        //return false; //commented for now to keep popup behaviour along with hyperlink;
                    });
                });
    });

</script>

**Comment by jrobinso** [comment]
I'm not sure what your question is.  If you have CORS errors they will have to be addressed on the server.   I tried your config and get a CORS error on the fasta index.    Checking the response there are no CORS headers present.   There is lots of information on CORS available on the web.
